leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/impl/CaliforniumLwM2mClientRequestSender.java

```diff
@@ -163,12 +164,12 @@ public void onTimeout() {
 
         @Override
         public void onCancel() {
-            errorCallback.onError(new RequestFailedException("Canceled request"));
+            errorCallback.onError(new RequestCanceledException("Canceled request"));
```

Comment:
I'm not sure we need a message parameter for `RequestCanceledException` ? (same thing for sender at server side)